skglm/datafits/single_task.py

```diff
@@ -774,3 +773,114 @@ def _get_indptr(self, vals, indices):
         indptr.append(n_indices)
 
         return np.asarray(indptr, dtype=np.int64)
+
+
+class WeightedQuadratic(BaseDatafit):
+    """Weighted Quadratic datafit.
+
+    The datafit reads:
+
+    .. math:: 1 / (2 xx  n_"samples") ||y - Xw||_2 ^ 2
+
+    Attributes
+    ----------
+    Xty : array, shape (n_features,)
+        Pre-computed quantity used during the gradient evaluation.
+        Equal to ``X.T @ y``.
+    sample_weights : array, shape (n_train_samples,)
+        Weights for each sample.
+    Note
+    ----
+    The class is jit compiled at fit time using Numba compiler.
+    This allows for faster computations.
+    """
+
+    def __init__(self, sample_weights):
+        self.sample_weights = sample_weights
+        pass
+
+    def get_spec(self):
+        spec = (
+            ('Xty', float64[:]),
+            ('sample_weights', float64[:]),
+        )
+        return spec
+
+    def params_to_dict(self):
+        return {'sample_weights':self.sample_weights}
+
+    def get_lipschitz(self, X, y):
+        n_features = X.shape[1]
+        lipschitz = np.zeros(n_features, dtype=X.dtype)
+
+        for j in range(n_features):
+            lipschitz[j] = (self.sample_weights * X[:, j] ** 2).sum() / self.sample_weights.sum()
+
+        return lipschitz
+
+    def get_lipschitz_sparse(self, X_data, X_indptr, X_indices, y):
+        n_features = len(X_indptr) - 1
+        lipschitz = np.zeros(n_features, dtype=X_data.dtype)
+
+        for j in range(n_features):
+            nrm2 = 0.
+            for idx in range(X_indptr[j], X_indptr[j + 1]):
+                nrm2 += self.sample_weights[X_indices[idx]] * X_data[idx] ** 2
+
+            lipschitz[j] = nrm2 / self.sample_weights.sum()
+
+        return lipschitz
+
+    def initialize(self, X, y):
+        self.Xty = X.T @ (self.sample_weights * y)
```

Comment:
I would not call it Xty if it is not equal to `X.T @ y`. I suggest `wXty` where `w` stands for weighted